server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -448,7 +451,9 @@ public boolean hasChanges() {
         }
 
         public boolean masterNodeChanged() {
-            return newMasterNode != null;
+            String newMasterId = (newMasterNode != null) ? newMasterNode.getEphemeralId() : null;
+            String previousMasterId = (previousMasterNode != null) ? previousMasterNode.getEphemeralId() : null;
+            return Objects.equals(newMasterId, previousMasterId) == false;
```

Comment:
All of this is equivalent to the simpler
```
return Objects.equals(newMasterNode, previousMasterNode) == false;
```
(see equals implementation of DiscoveryNode)

Reply:
Right, thanks